spec-tests/src/test/java/org/candlepin/spec/content/OwnerContentSpecTest.java

```diff
@@ -0,0 +1,685 @@
+            .arches(content.getArches() + "-updated")
+            .metadataExpire(content.getMetadataExpire() + 10L);
+
+        ContentDTO updated = adminClient.ownerContent().updateContent(owner.getKey(), contentUpdate.getId(),
+            contentUpdate);
+
+        assertNotNull(updated);
+
+        // Impl note:
+        // We can't do a raw equality check since our generated content won't have a value for the
+        // uuid, created, or updated fields. We can cheat a bit and just use those on the returned
+        // DTO to test everything else with .equals
+        content.uuid(created.getUuid())
+            .created(created.getCreated())
+            .updated(created.getUpdated());
+
+        contentUpdate.uuid(updated.getUuid())
+            .created(updated.getCreated())
+            .updated(updated.getUpdated());
+
+        assertEquals(content, created);
+        assertEquals(contentUpdate, updated);
+        assertNotEquals(created, updated);
+
+        ContentDTO fetched = adminClient.ownerContent().getOwnerContent(owner.getKey(), updated.getId());
+        assertNotNull(fetched);
+
+        // Same deal here, but we've already set the fields to what they should be upstream
+        assertEquals(contentUpdate, fetched);
+    }
+
+    @Test
+    @DisplayName("should not allow updating critical fields")
+    public void shouldNotAllowUpdatingCriticalFields() throws Exception {
+        // At the time of writing, this is only the UUID and ID fields
+
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+
+        ContentDTO created = adminClient.ownerContent().createContent(owner.getKey(), Content.random());
+        assertNotNull(created);
+
+        // Cache the ID since we're about to clobber it
+        String id = created.getId();
+
+        // These fields should be silently ignored during an update
+        ContentDTO update = Content.copy(created)
+            .uuid("updated_uuid")
+            .id("updated_id");
+
+        ContentDTO updated = adminClient.ownerContent().updateContent(owner.getKey(), id, update);
+        assertNotNull(updated);
+
+        // Both DTOs should be identical yet, since the fields changed aren't changeable
+        assertEquals(created, updated);
+
+        ContentDTO fetched = adminClient.ownerContent().getOwnerContent(owner.getKey(), id);
+        assertNotNull(fetched);
+
+        // Same here; should still be identical
+        assertEquals(created, fetched);
+        assertEquals(updated, fetched);
+    }
+
+    @Test
+    @DisplayName("should not allow updating locked content")
+    public void shouldNotAllowUpdatingLockedContent() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+
+        // TODO: This test cannot be completed until the upstream bits are available, since we can no
+        // longer query or set the locked flag via API
```

Comment:
In addition to the upstream bits, you also need the ability of sending a custom JSON payload for the content PUT request here, since the locked flag isn't on the DTO. I think we need to extend the scope of the task 'Ability to assert on JSON response' to also include 'Ability to send custom JSON payload to an endpoint'.

Until those are done though it might be better to mark this with the `@Disabled("Depends on X and Y features being ported to java spec tests")` annotation, to skip it entirely during runs